cmd/odf/devpreview/devpreview.go

```diff
@@ -0,0 +1,92 @@
+`
+	// AllowFlag is required to run developer preview comamands.
+	AllowFlag = "allow-developer-preview"
+
+	// We want to include the command long description in the usage message so when
+	// the required --allow-developer-preview flag is missing, we show the developer
+	// preview note. The only way seems to be to redefine the help and usage
+	// templates.
+
+	// Copy of cobra defaultHelpTemplate always using only the .Short descripiton.
+	HelpTemplate = `{{with .Short}}{{. | trim}}
+
+{{end}}{{if or .Runnable .HasSubCommands}}{{.UsageString}}{{end}}`
+
+	// Copy of cobra defaulUsageTemplate displaying also the .Long descripiton.
+	UsageTemplate = `{{with .Long}}{{. | trim}}
+
+{{end}}Usage:{{if .Runnable}}
+  {{.UseLine}}{{end}}{{if .HasAvailableSubCommands}}
+  {{.CommandPath}} [command]{{end}}{{if gt (len .Aliases) 0}}
+
+Aliases:
+  {{.NameAndAliases}}{{end}}{{if .HasExample}}
+
+Examples:
+{{.Example}}{{end}}{{if .HasAvailableSubCommands}}{{$cmds := .Commands}}{{if eq (len .Groups) 0}}
+
+Available Commands:{{range $cmds}}{{if (or .IsAvailableCommand (eq .Name "help"))}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{else}}{{range $group := .Groups}}
+
+{{.Title}}{{range $cmds}}{{if (and (eq .GroupID $group.ID) (or .IsAvailableCommand (eq .Name "help")))}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{if not .AllChildCommandsHaveGroup}}
+
+Additional Commands:{{range $cmds}}{{if (and (eq .GroupID "") (or .IsAvailableCommand (eq .Name "help")))}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{end}}{{end}}{{if .HasAvailableLocalFlags}}
+
+Flags:
+{{.LocalFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasAvailableInheritedFlags}}
+
+Global Flags:
+{{.InheritedFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasHelpSubCommands}}
+
+Additional help topics:{{range .Commands}}{{if .IsAdditionalHelpTopicCommand}}
+  {{rpad .CommandPath .CommandPathPadding}} {{.Short}}{{end}}{{end}}{{end}}{{if .HasAvailableSubCommands}}
+
+Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
+`
+)
+
+// Configure a command as a developer preview comamnd. Set the given command
+// help and usage template, add a required `--allow-developer-preview` flag, and
+// configure the command short and long descriptions. If the command has
+// children commands their short and long descriptions are configured as well.
+func Configure(cmd *cobra.Command) {
+	// Templates are inherited by children commands.
+	cmd.SetUsageTemplate(UsageTemplate)
+	cmd.SetHelpTemplate(HelpTemplate)
+
+	// Persitent flags are inherited by children commands.
+	cmd.PersistentFlags().Bool(AllowFlag, false, "mandatory flag to run developer preview commands")
+	cmd.MarkPersistentFlagRequired(AllowFlag)
+
+	setDescriptions(cmd)
+}
+
+func setDescriptions(cmd *cobra.Command) {
+	// Set the short and long descripption of this command.
```

Comment:
typo, "description"

Reply:
Comment removed when simplifying the code.